primitives/src/traits/market_transition_api.rs

```diff
@@ -0,0 +1,322 @@
+            }
+        )* );
+        collective_result
+    }
+    fn on_report(market_id: &MI) -> ResultWithWeightInfo<DispatchResult> {
+        let mut collective_result = ResultWithWeightInfo::new(Ok(()), Weight::zero());
+        for_tuples!( #(
+            let result = Tuple::on_report(market_id);
+            collective_result.result = result.result;
+            collective_result.weight = collective_result.weight.saturating_add(result.weight);
+            if collective_result.result.is_err() {
+                return collective_result;
+            }
+        )* );
+        collective_result
+    }
+    fn on_dispute(market_id: &MI) -> ResultWithWeightInfo<DispatchResult> {
+        let mut collective_result = ResultWithWeightInfo::new(Ok(()), Weight::zero());
+        for_tuples!( #(
+            let result = Tuple::on_dispute(market_id);
+            collective_result.result = result.result;
+            collective_result.weight = collective_result.weight.saturating_add(result.weight);
+            if collective_result.result.is_err() {
+                return collective_result;
+            }
+        )* );
+        collective_result
+    }
+    fn on_resolution(market_id: &MI) -> ResultWithWeightInfo<DispatchResult> {
+        let mut collective_result = ResultWithWeightInfo::new(Ok(()), Weight::zero());
+        for_tuples!( #(
+            let result = Tuple::on_resolution(market_id);
+            collective_result.result = result.result;
+            collective_result.weight = collective_result.weight.saturating_add(result.weight);
+            if collective_result.result.is_err() {
+                return collective_result;
+            }
+        )* );
+        collective_result
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use frame_support::pallet_prelude::DispatchError;
+
+    const DEFAULT_ERROR: DispatchResult = Err(DispatchError::Other("unimportant"));
+    const ONE: Weight = Weight::from_all(1);
+    const TWO: Weight = Weight::from_all(2);
+    const THREE: Weight = Weight::from_all(3);
+
+    struct ExecutionPath;
+    impl MarketTransitionApi<u128> for ExecutionPath {
+        fn on_proposal(_market_id: &u128) -> ResultWithWeightInfo<DispatchResult> {
+            panic!("on_proposal");
+        }
+        fn on_activation(_market_id: &u128) -> ResultWithWeightInfo<DispatchResult> {
+            panic!("on_activation");
+        }
+        fn on_closure(_market_id: &u128) -> ResultWithWeightInfo<DispatchResult> {
+            panic!("on_closure");
+        }
+        fn on_report(_market_id: &u128) -> ResultWithWeightInfo<DispatchResult> {
+            panic!("on_report");
+        }
+        fn on_dispute(_market_id: &u128) -> ResultWithWeightInfo<DispatchResult> {
+            panic!("on_dispute");
+        }
+        fn on_resolution(_market_id: &u128) -> ResultWithWeightInfo<DispatchResult> {
+            panic!("on_resolution");
+        }
+    }
+
+    struct SucessPath;
```

Comment:
Should be called `SuccessPath`.

Reply:
[SuccessPath](https://github.com/zeitgeistpm/zeitgeist/pull/1281/commits/427b7cedde5a95c2eb002ed3684678aa8f0812f2)